tests/test_archive_querier.py

```diff
@@ -290,6 +294,56 @@ def test_paginate_many_records_single_time_bucket(table_maker, querier):
     evaluate_time_based_results(results, 150)
 
 
+def test_paginate_few_records_single_bucket_no_empty_page(table_maker,
+                                                          querier):
+    records = []
+    # Fill one bucket with 2x MAX_RESULTS,
+    # but we only want the first 2 of the last 3 records
+    interval = DatalakeRecord.TIME_BUCKET_SIZE_IN_MS / MAX_RESULTS / 2
+    very_end = DatalakeRecord.TIME_BUCKET_SIZE_IN_MS
+    for start in range(0, very_end, interval):
+        end = start + interval
+        records += create_test_records(start=start, end=end, what='foo')
+    table_maker(records)
+    results = get_multiple_pages(
+        querier.query_by_time,
+        [very_end - 3 * interval + 1,
+         very_end - interval - 1, 'foo'],
+        True)
```

Comment:
"The first 2 of the last 3 records?" Dizzying. Why can't we just target the last record? `start` should land just where you want it. Then this query becomes something like:

```
results = get_multiple_pages(querier.query_by_time, [start, start + 1, 'foo'])
```

I tried this. It does seem to reproduce the error (along with the test condition I proposed above).